apps/glusterfs/app.go

```diff
@@ -98,7 +114,20 @@ func NewApp(configIo io.Reader) *App {
 		logger.Warning("Unable to open database.  Retrying using read only mode")
 
 		// Try opening as read-only
-		app.db, err = bolt.Open(dbfilename, 0666, &bolt.Options{
+		app.db, err = bolt.Open(dbfilename, 0600, &bolt.Options{
+			ReadOnly: true,
+		})
+		if err != nil {
+			logger.LogError("Unable to open database: %v", err)
+			return nil
+		}
+		app.dbReadOnly = true
+	} else if nil != checkBoltDbWrite(app) {
+		logger.Warning("write failed. Retrying using read only mode")
+                app.db.Close()
+
+		// Try opening as read-only
+		app.db, err = bolt.Open(dbfilename, 0600, &bolt.Options{
```

Comment:
I think we should do 666 here. It worked for me. can you add more detail why not 600?